### ISSUE TITLE
How to save the training log of lightgbm?

### ISSUE THREAD

**Issue by hepengfei-ml** [microsoft/LightGBM#4764]
How to save the training log of lightgbm?

**Comment by jameslamb** [comment]
Hi @hepengfei-ml , thanks for your interest in LightGBM.

I'm not sure what you mean by "the training log". Can you please add more details, so we can give you the most specific and helpful answer possible?

**Comment by hepengfei-ml** [comment]
I want to save the following log in  my log


[LightGBM] [Info] Construct bin mappers from text data time 1.18 seconds
[LightGBM] [Info] Number of positive: 2240897, number of negative: 10005005
[LightGBM] [Warning] Auto-choosing row-wise multi-threading, the overhead of testing was 1.545772 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 5727
[LightGBM] [Info] Number of data points in the train set: 12245902, number of used features: 65
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.182992 -> initscore=-1.496209
[LightGBM] [Info] Start training from score -1.496209

**Comment by shiyu1994** [comment]
@hepengfei-ml Thanks for using LightGBM. LightGBM don't provide any internal method to save the output log. But if you are using python package, you can save the loss and metric of every iteration in a dictionary. Please check the `record_evaluation` callback function
https://lightgbm.readthedocs.io/en/latest/Python-API.html#callbacks
If you want to save the full log into your file, you may use some output redirection of your operation system.

**Comment by StrikerRUS** [comment]
@hepengfei-ml 

> If you want to save the full log into your file, you may use some output redirection of your operation system.

If you're using Python-package, it can be easily done with the help of `register_logger()` function: https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.register_logger.html.
Here is an example:
https://github.com/microsoft/LightGBM/blob/da98f24711a2faab17f94e5b2a636e6609c93fa6/tests/python_package_test/test_utilities.py#L10-L23

**Comment by gongyulian** [comment]
excellent !  thank you !!!

**Comment by shiyu1994** [comment]
@hepengfei-ml Will close this issue. If you have any further question, please feel free to reopen it. Thansk!
